sdk/common/src/main/scala/org/typelevel/otel4s/sdk/Resource.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk
+
+import cats.Show
+import cats.implicits.showInterpolator
+import org.typelevel.otel4s.Attribute
+import org.typelevel.otel4s.sdk.BuildInfo
+import org.typelevel.otel4s.semconv.resource.attributes.ResourceAttributes._
+
+/** [[Resource]] serves as a representation of a resource that captures
+  * essential identifying information regarding the entities associated with
+  * reported signals, such as statistics or traces.
+  * @param attributes
+  *   \- a collection of [[Attribute]]s
+  * @param schemaUrl
+  *   \- an optional schema URL
+  */
+case class Resource(
+    attributes: Attributes,
+    schemaUrl: Option[String] = None
+) {
+
+  /** Merges [[Resource]] into another [[Resource]]. If the same attribute
+    * exists in both resources, the attribute in the other [[Resource]] will be
+    * used.
+    * @param other
+    *   \- the other [[Resource]] to merge into.
+    * @return
+    *   a new [[Resource]] with the merged attributes.
+    */
+  def mergeInto(other: Resource): Resource = {
```

Comment:
Definitely it's not a `Monoid` because it doesn't satisfy the association law. Depending on the order, we will have different `schemaUrl` values.

Reply:
> Depending on the order

Do you mean commutativity?

Reply:
> Do you mean commutativity?

Reply:
Not sure about `schemaUrl` maybe it would be better to return an error for this case. For now I used the same logic that was described in Java SDK, but it could be `Either[MergeConflictError, Resource]` as well.

Reply:
> but it could be `Either[MergeConflictError, Resource]` as well.

Hmm. Alternatively, we could have three states, `Some`, `None`, and `Undefined`.

Reply:
I am assuming that we might have other validations in the merge method, so I have chosen the `Either` option. def mergeInto(other: Resource): Either[ResourceInitiationError, Resource]`.